packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -104,47 +119,49 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
         if (_target == Predeploys.CROSS_L2_INBOX) revert MessageTargetCrossL2Inbox();
         if (_target == Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER) revert MessageTargetL2ToL2CrossDomainMessenger();
 
-        bytes memory data = abi.encodeCall(
-            L2ToL2CrossDomainMessenger.relayMessage,
-            (_destination, block.chainid, messageNonce(), msg.sender, _target, _message)
-        );
-        assembly {
-            log0(add(data, 0x20), mload(data))
-        }
+        emit SentMessage(_destination, _target, messageNonce(), msg.sender, _message);
+
         msgNonce++;
     }
 
-    /// @notice Relays a message that was sent by the other CrossDomainMessenger contract. Can only be executed via
-    ///         cross-chain call from the other messenger OR if the message was already received once and is currently
-    ///         being replayed.
-    /// @param _destination Chain ID of the destination chain.
-    /// @param _source      Chain ID of the source chain.
-    /// @param _nonce       Nonce of the message being relayed.
-    /// @param _sender      Address of the user who sent the message.
-    /// @param _target      Address that the message is targeted at.
-    /// @param _message     Message payload to call target with.
+    /// @notice Relays a message that was sent by the other L2ToL2CrossDomainMessenger contract. Can only be executed
+    ///         via cross chain call from the other messenger OR if the message was already received once and is
```

Comment:
weird question, but, could we send and relay a message on the same domain? does this open up any edge-cases?
asking because the natspec should also change if so.
this might or might not be possible because the DependencySet should "kinda" include its own chainid.

Reply:
So Interop does allow sending messages to the same domain as the chain id is apart of the dependency set by default. This allows contracts to bring logs on chain which is net new. I would expect there to be some cool usages of this.

However not sure what use cases sending a CALL to the same domain would have. The only thing gained is having `msg.sender == L2ToL2CrossDomainMessenger` for the call.